src/system.rs

```diff
@@ -272,20 +271,29 @@ impl PendingTypes {
         type_id + type_store.types_by_id.len()
     }
 
-    /// Adds the [CoreTypeDefinition] and the associated ion_type in the [PendingTypes] and returns the [TypeId] for it
+    /// Adds the [CoreISLTypeDefinition] in the [PendingTypes] and returns the [TypeId] for it
     /// If the given name already exists in the [TypeStore] or [PendingTypes] it returns the associated [TypeId]
-    pub fn add_core_type(&mut self, ion_type: IonType, type_store: &mut TypeStore) -> TypeId {
-        let core_type = &format!("{:?}", ion_type);
-        if let Some(exists) = self.core_type_ids_by_ion_type.get(core_type) {
+    pub fn add_core_type(
```

Comment:
If we're going to load these types lazily, why not put them in the same hash map as the imported types? They are basically an import (since they come from outside of the schema) even if the import is implicit.

Also, since the core types are always available to all schemas, would it make sense to eagerly load them instead?

Reply:
Preloading all the builtin types to `TypeStore`.